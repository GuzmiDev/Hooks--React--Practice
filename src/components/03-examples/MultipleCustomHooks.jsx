import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { useFetch } from "../../hooks/useFetch";
import "../02-useEffect/effects.css";

export const MultipleCustomHooks = () => {
  const { counter, increment, decrement } = useCounter(1);

  const { loading, data } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );
  const { author, quote } = !!data && data[0];

  return (
    <div>
      <h1>BreakingBad Quotes</h1>
      <hr />

      <p>Frase numero: {counter}</p>

      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="text-end blockquote">
          <p>{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}

      <button className="btn btn-primary m-1" type="button" onClick={decrement}>
        Anterior frase
      </button>
      <button className="btn btn-primary m-1" type="button" onClick={increment}>
        Siguiente frase
      </button>
    </div>
  );
};
