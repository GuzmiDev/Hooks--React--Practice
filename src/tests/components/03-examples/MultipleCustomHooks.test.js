import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en <MultipleCustomHooks />", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => {},
      decrement: () => {},
    });
  });

  test("debe de mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar la informaciÃ³n", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Fernando",
          quote: "Hola Mundo",
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find("#quote").text().trim()).toBe("Hola Mundo");
    expect(wrapper.find("#author").text().trim()).toBe("Fernando");
  });
});
