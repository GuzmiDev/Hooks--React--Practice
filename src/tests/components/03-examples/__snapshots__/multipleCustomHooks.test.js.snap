// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrar la informaci√≥n 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <p>
    Frase numero: 
    1
  </p>
  <blockquote
    className="text-end blockquote"
  >
    <p
      id="quote"
    >
      Hola Mundo
    </p>
    <footer
      className="blockquote-footer"
      id="author"
    >
      Fernando
    </footer>
  </blockquote>
  <button
    className="btn btn-primary m-1"
    onClick={[Function]}
    type="button"
  >
    Anterior frase
  </button>
  <button
    className="btn btn-primary m-1"
    onClick={[Function]}
    type="button"
  >
    Siguiente frase
  </button>
</div>
`;

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <p>
    Frase numero: 
    1
  </p>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary m-1"
    onClick={[Function]}
    type="button"
  >
    Anterior frase
  </button>
  <button
    className="btn btn-primary m-1"
    onClick={[Function]}
    type="button"
  >
    Siguiente frase
  </button>
</div>
`;
